mlflow/genai/judges/instructions_judge/__init__.py

```diff
@@ -184,6 +195,21 @@ def __call__(
                 error_code=INVALID_PARAMETER_VALUE,
             )
 
+        # Extract fields from trace if needed (when not using {{ trace }} template)
+        if trace is not None and self._TEMPLATE_VARIABLE_TRACE not in self.template_variables:
+            extracted = extract_evaluation_fields_from_trace(trace)
+
+            # Use extracted values if not already provided
+            if inputs is None and self._TEMPLATE_VARIABLE_INPUTS in self.template_variables:
+                inputs = extracted.inputs
+            if outputs is None and self._TEMPLATE_VARIABLE_OUTPUTS in self.template_variables:
+                outputs = extracted.outputs
+            if (
+                expectations is None
+                and self._TEMPLATE_VARIABLE_EXPECTATIONS in self.template_variables
+            ):
+                expectations = extracted.expectations
```

Comment:
@BenWilson2 I considered the case where `instructions` contains variables `{{trace}}` and `{{inputs}}`.

If the instructions judge is called with `__call___(trace=trace, inputs=inputs)`, it seems that `inputs` are never passed to the prompt because we use the trace-based evaluation logic - https://github.com/mlflow/mlflow/pull/17571/files#diff-2cd05ad709d0d3fdf0757afab20f023ee8b12ad82308ed898b1d8c4368e32fe3R291-R308.

Can we fix this and add a test case?